core/src/transport/upgrade.rs

```diff
@@ -299,7 +299,7 @@ impl<T> Multiplexed<T> {
         T::Dial: Send + 'static,
         T::ListenerUpgrade: Send + 'static,
         T::Error: Send + Sync,
-        M: StreamMuxer + Send + Sync + 'static,
+        M: StreamMuxer + Send + 'static,
         M::Substream: Send + Unpin + 'static,
         M::Error: Send + Sync + 'static,
```

Comment:
Could we also drop the other `Sync` requirements here?

Reply:
Maybe. I'd say we do it in a different PR though :)